chromium_src/chrome/browser/ui/views/first_run_dialog.cc

```diff
@@ -3,18 +3,127 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "chrome/browser/ui/views/first_run_dialog.h"
+
+#include <string>
+
+#include "base/bind.h"
+#include "base/run_loop.h"
 #include "brave/grit/brave_generated_resources.h"
+#include "build/build_config.h"
+#include "chrome/browser/first_run/first_run.h"
+#include "chrome/browser/first_run/first_run_dialog.h"
+#include "chrome/browser/metrics/metrics_reporting_state.h"
+#include "chrome/browser/platform_util.h"
+#include "chrome/browser/shell_integration.h"
+#include "chrome/browser/ui/browser_dialogs.h"
+#include "chrome/browser/ui/ui_features.h"
+#include "chrome/browser/ui/views/chrome_layout_provider.h"
+#include "chrome/common/url_constants.h"
 #include "chrome/grit/chromium_strings.h"
 #include "chrome/grit/generated_resources.h"
+#include "components/crash/core/app/breakpad_linux.h"
+#include "components/crash/core/app/crashpad.h"
+#include "components/strings/grit/components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/views/border.h"
+#include "ui/views/controls/button/checkbox.h"
+#include "ui/views/controls/link.h"
+#include "ui/views/layout/box_layout.h"
+#include "ui/views/layout/grid_layout.h"
+#include "ui/views/widget/widget.h"
+#include "ui/views/window/dialog_delegate.h"
+
+namespace first_run {
+
+void ShowFirstRunDialog(Profile* profile) {
+#if defined(OS_MAC)
+  if (base::FeatureList::IsEnabled(features::kViewsFirstRunDialog))
+    ShowFirstRunDialogViews(profile);
+  else
+    ShowFirstRunDialogCocoa(profile);
+#else
+  ShowFirstRunDialogViews(profile);
+#endif
+}
+
+void ShowFirstRunDialogViews(Profile* profile) {
+  FirstRunDialog::Show(profile);
+}
+
+}  // namespace first_run
+
+// static
+void FirstRunDialog::Show(Profile* profile) {
+  FirstRunDialog* dialog = new FirstRunDialog(profile);
+  views::DialogDelegate::CreateDialogWidget(dialog, NULL, NULL)->Show();
```

Comment:
Why don't you use the `unique_ptr` constructor? This looks like a memory leak

Reply:
No leak - it's owned by Widget. It will be destroyed when dialog is done.
https://source.chromium.org/chromium/chromium/src/+/main:ui/views/widget/widget_delegate.cc;l=450

I assume this `FirstRunDialog` was created quite long time ago before `unique_ptr` version of `CreateDialogWidget` is added. Of course, we can update but it's dtor is in private section. (need to move it to public to use `unique_ptr` and need patching.) I think it's fine to go as-is because it's no-leak.

Reply:
Sure thing thanks for the explanation